legacy/src/main/java/org/opensearch/sql/legacy/executor/format/DateFieldFormatter.java

```diff
@@ -152,15 +152,27 @@ private Date parseDateString(List<String> formats, String columnOriginalDate) {
         switch (columnFormat) {
           case "date_optional_time":
           case "strict_date_optional_time":
-            parsedDate =
-                DateUtils.parseDate(
-                    columnOriginalDate,
-                    FORMAT_DOT_OPENSEARCH_DASHBOARDS_SAMPLE_DATA_LOGS_EXCEPTION,
-                    FORMAT_DOT_OPENSEARCH_DASHBOARDS_SAMPLE_DATA_FLIGHTS_EXCEPTION,
-                    FORMAT_DOT_OPENSEARCH_DASHBOARDS_SAMPLE_DATA_FLIGHTS_EXCEPTION_NO_TIME,
-                    FORMAT_DOT_OPENSEARCH_DASHBOARDS_SAMPLE_DATA_ECOMMERCE_EXCEPTION,
-                    FORMAT_DOT_DATE_AND_TIME,
-                    FORMAT_DOT_DATE);
+            // It's possible to have date stored in second / millisecond form without explicit
+            // format hint.
+            // Parse it on a best-effort basis.
```

Comment:
Could you check if anything in OpenSearch's `DateFormatters` can be reused?

Reply:
I dont see there is a suitable method I can call to detect and parse it on best effort basis, if we decide the use `DataFormatters`, the logic will turns into following:

```
              try {
                parsedDate = Date.from(DateFormatters.from(
                                DateFormatter.forPattern("epoch_millis").parse(columnOriginalDate))
                        .toInstant());
              } catch (DateTimeException ex) {
                // Fallback to epoch_second
                try {
                  parsedDate = Date.from(DateFormatters.from(
                                  DateFormatter.forPattern("epoch_second").parse(columnOriginalDate))
                          .toInstant());
                } catch (DateTimeException innerEx) {
                  parsedDate =
                          DateUtils.parseDate(
                                  columnOriginalDate,
                                  FORMAT_DOT_OPENSEARCH_DASHBOARDS_SAMPLE_DATA_LOGS_EXCEPTION,
                                  FORMAT_DOT_OPENSEARCH_DASHBOARDS_SAMPLE_DATA_FLIGHTS_EXCEPTION,
                                  FORMAT_DOT_OPENSEARCH_DASHBOARDS_SAMPLE_DATA_FLIGHTS_EXCEPTION_NO_TIME,
                                  FORMAT_DOT_OPENSEARCH_DASHBOARDS_SAMPLE_DATA_ECOMMERCE_EXCEPTION,
                                  FORMAT_DOT_DATE_AND_TIME,
                                  FORMAT_DOT_DATE);
                }
                
```
Because I need the try-catch to perform the parsing logic fallback from millisecond, second and eventually a normal date String.
Which I don't think that is ideal from the performance perspective, as this will run against every row being resulted.